File: javascript/ts/org/noear/socketd/transport/core/Message.ts
import {Entity} from "./Entity";

/**
 * 消息
 *
 * @author noear
 * @since 2.0
 * @class
 */
export interface Message extends Entity {
    /**
     * 是否为请求
     * @return {boolean}
     */
    isRequest(): boolean;

    /**
     * 是否为订阅
     * @return {boolean}
     */
    isSubscribe(): boolean;

    /**
     * 获取消息流Id（用于消息交互、分片）
     * @return {string}
     */
    sid(): string;

    /**
     * 获取消息主题
     * @return {string}
     */
    topic(): string;

    /**
     * 获取消息实体（有时需要获取实体）
     * @return {*}
     */
    entity(): Entity;
}

File: swift/socketd/socketd/src/socketd/transport/core/Message.swift
//
//  Message.swift
//  socketd
//
//  Created by noear on 2023/12/27.
//

import Foundation

/**
 * 消息
 *
 * @author noear
 * @since 2.0
 */
protocol Message : Entity{
    /**
     * 是否为请求
     */
    func isRequest() -> Bool;
    
    /**
     * 是否为订阅
     */
    func isSubscribe() ->Bool;
    
    /**
     * 获取消息流Id（用于消息交互、分片）
     */
    func sid() -> String;
    
    /**
     * 获取消息事件
     */
    func event()->String;
    
    /**
     * 获取消息实体（有时需要获取实体）
     */
    func entity()->Entity;
    
}
